client/components/PetAdoptionForm.js

```diff
@@ -0,0 +1,128 @@
+import React, {useState} from "react"
+
+import ErrorList from "./ErrorsList"
+
+const PetAdoptionForm = (props) => {
+  const [errors, setErrors] = useState({})
+  const [adoption, setAdoption] = useState({
+    name: "",
+    phoneNumber: "",
+    email: "",
+    homeStatus: "",
+    adoptablePetId: props.petId,

```

Comment:
You need to be very careful when using something from props to set the initial value in a piece of state.  Since this will only be used the first time the page renders, if props.petId was to change it would not be reflected in state, because that would not cause useState to run again.  Instead you could attach the adoptablePetId property to the object right before sending it in the POST request